### PR TITLE
Read all pixels then write all pixels to avoid virtual call overhead

### PR DESCRIPTION
**Pull request by Gillibald**
## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

### CODE REVIEW

**Comment by maxkatz6** [comment]
cc @kekekeks

File: src/Avalonia.Base/Media/Imaging/PixelFormatTranscoder.cs [hunk 1]
```diff
@@ -15,140 +17,16 @@ internal static unsafe class PixelFormatTranscoder
         PixelFormat destFormat,
         AlphaFormat destAlphaFormat)
     {
-        var reader = GetReader(srcFormat);
-        var writer = GetWriter(destFormat);
+        var pixelCount = srcSize.Width * srcSize.Height;
+        var bufferSize = pixelCount * Marshal.SizeOf<Rgba8888Pixel>();
+        var blob = new UnmanagedBlob(bufferSize);
```
Comment (kekekeks) [inline src/Avalonia.Base/Media/Imaging/PixelFormatTranscoder.cs:22]: Use `using var` instead of manual Dispose

**Review by kekekeks — APPROVED** [review]

File: src/Avalonia.Base/Media/Imaging/PixelFormatTranscoder.cs (hunk 1 above)
Comment (Gillibald) [reply src/Avalonia.Base/Media/Imaging/PixelFormatTranscoder.cs:22]: UnmanagedBlob isn't IDisposable

Comment (kekekeks) [reply src/Avalonia.Base/Media/Imaging/PixelFormatTranscoder.cs]: Should probably make it to implement IDisposable, it has Dispose method and the entire finalizer pattern.

**Pull request merged**